src/MetricsUploader/MetricsUploaderWindows.cpp

```diff
@@ -108,9 +108,11 @@ ErrorMessageOr<MetricsUploader> MetricsUploader::CreateMetricsUploader(std::stri
   // set up connection and create a client
   Result result = setup_connection_addr();
   if (result != kNoError) {
-    Result result = shutdown_connection_addr();
-    if (result != kNoError) {
-      ERROR("Error while closing connection: %s", GetErrorMessage(result));
+    if (result != kCannotOpenConnection) {
```

Comment:
I think these two `if`s can be combined.

Reply:
And one question: Is `shutdown_connection_addr()` the inverse of `setup_connection_addr()`? If yes, then I would say it's untypical to call shutdown when setup didn't succeed. That should not be necessary.

Reply:
There also some additional work in SetupConnection that could fail. Because of it I'm checking the error type and closing connection if the error occurred. 
I've renamed the variable for the ShutdownConnection() result as I used the same name for 2 different results.

Reply:
Regarding the first comment: did you mean something like
```
if (result != kNoError && result != kCannotOpenConnection) {
  shutdown
}
if (result != kNoError) {
  FreeLibrary()
}
```

Reply:
Ah I'm sorry. I didn't see the `FreeLibrary`. I guess in that case you solution is fine,

Reply:
No worries, thank you for the review!